plugins/aws/elasticache/redisClusterEncryptionAtRest.spec.js

```diff
@@ -0,0 +1,270 @@
+            "CreationDate": "2020-12-15T01:16:53.045000+05:00",
+            "Enabled": true,
+            "Description": "Default master key that protects my Glue data when no other key is defined",
+            "KeyUsage": "ENCRYPT_DECRYPT",
+            "KeyState": "Enabled",
+            "Origin": "AWS_KMS",
+            "KeyManager": "AWS",
+            "CustomerMasterKeySpec": "SYMMETRIC_DEFAULT",
+            "EncryptionAlgorithms": [
+                "SYMMETRIC_DEFAULT"
+            ]
+        }
+    }
+];
+
+const listKeys = [
+    {
+        "KeyId": "0604091b-8c1b-4a55-a844-8cc8ab1834d9",
+        "KeyArn": "arn:aws:kms:us-east-1:000011112222:key/c4750c1a-72e5-4d16-bc72-0e7b559e0250"
+    }
+]
+
+const createCache = (clusters, keys, describeReplicationGroups, describeKey, clustersErr, keysErr, describeKeyErr, describeReplicationGroupsErr) => {
+    var keyId = (clusters && clusters.length && clusters[0].KmsKeyId) ? clusters[0].KmsKeyId.split('/')[1] : null;
+    var ReplicationGroupId = (clusters && clusters.length) ? clusters[0].ReplicationGroupId: null;
+    return {
+        elasticache: {
+            describeCacheClusters: {
+                'us-east-1': {
+                    err: clustersErr,
+                    data: clusters
+                },
+            },
+            describeReplicationGroups: {
+                'us-east-1': {
+                    [ReplicationGroupId]: {
+                        data: describeReplicationGroups,
+                        err: describeReplicationGroupsErr
+                    }
+                }
+            }
+        },
+        kms: {
+            listKeys: {
+                'us-east-1': {
+                    data: keys,
+                    err: keysErr
+                }
+            },
+            describeKey: {
+                'us-east-1': {
+                    [keyId]: {
+                        err: describeKeyErr,
+                        data: describeKey
+                    },
+                },
+            },
+        },
+    };
+};
+
+describe('redisClusterEncryptionAtRest', function () {
+    describe('run', function () {
+        it('should PASS if Redis Cluster at-rest is encrypted with desired encryption level', function (done) {
+            const cache = createCache(describeCacheClusters[0], listKeys, describeReplicationGroups[0], describeKey[0]);
+            redisClusterEncryptionAtRest.run(cache, { ec_atrest_desired_encryption_level: 'awscmk' }, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should FAIL if Redis Cluster at-rest is not encrypted with desired encryption level', function (done) {
+            const cache = createCache([describeCacheClusters[1]],listKeys, describeReplicationGroups[1], describeKey[1]);
```

Comment:
describeReplicationGroups[1] does nt have kms key arn. is that fine?

Reply:
Yes because if "AtRestEncryptionEnabled" is true, it means that AWS-managed encryption is being used